lib/lndclient/LndClient.ts

```diff
@@ -352,15 +352,16 @@ class LndClient extends SwapClient {
     return this.unaryCall<lndrpc.ListChannelsRequest, lndrpc.ListChannelsResponse>('listChannels', new lndrpc.ListChannelsRequest());
   }
 
-  public getRoutes =  async (amount: number, destination: string): Promise<lndrpc.Route[]> => {
+  public getRoutes =  async (amount: number, destination: string, cltvDelta?: number): Promise<lndrpc.Route[]> => {
     const request = new lndrpc.QueryRoutesRequest();
+    const finalCltvDelta = cltvDelta || this.cltvDelta;
```

Comment:
Why this.cltvDelta should be used here?
argument should not be optional IMHO

Reply:
I originally thought about making the argument mandatory, but the second usage of `getRoutes` happens within the context of maker at the beginning of the deal execution process when [taker queries routes to maker](https://github.com/ExchangeUnion/xud/blob/master/lib/swaps/Swaps.ts#L254).

At this point taker has the following information:
```
const makerOrder = {
  id: 'fb1708f0-90d7-11e9-8836-9bf81440c6fa',
  pairId: 'LTC/BTC',
  price: 0.1,
  quantity: 10000,
  isBuy: true,
  peerPubKey:
   '038d48a01d25735cd7bd26ac8cbd7583e4c202718848ab63b08b378dc840a24e81',
  createdAt: 1560759291688,
  initialQuantity: 10000,
};

const takerOrder = {
  pairId: 'LTC/BTC',
  price: 0.1,
  quantity: 10000,
  isBuy: false,
  localId: 'fd6da320-90d7-11e9-9e9d-0f0eb626ffe9',
  hold: 0,
  id: 'fd6da320-90d7-11e9-9e9d-0f0eb626ffe9',
  initialQuantity: 10000,
  createdAt: 1560759295570,
};
```

What would you set the `finalCltvDelta` to? Furthermore, the value of `takerToMakerRoutes` is being used as a light sanity check (error when 0 routes found). Alternative option is to remove the `getRoutes` check from the taker side.

Reply:
In order to be more explicit I changed `getRoutes` to require `finalCltvDelta`.

Reply:
Fwiw I found it a bit odd to pass an object's own property to one of its method calls as in `routes = await swapClient.getRoutes(makerAmount, destination, swapClient.cltvDelta);`.

Another way to consider writing this is:

```
public getRoutes =  async (amount: number, destination: string, finalCltvDelta = this.cltvDelta) => {
```

That way it's clear what the value defaults to if its not specified. Ultimately not a big deal though so it definitely be left as is.

Reply:
After thinking about it for a while I agree, it's weird to pass an object's own property to one of its method calls. I changed it back.